addons/payment_stripe/models/payment.py

```diff
@@ -48,17 +48,24 @@ def stripe_form_generate_values(self, tx_values):
             'cancel_url': urls.url_join(base_url, StripeController._cancel_url) + '?reference=%s' % tx_values['reference'],
             'customer_email': tx_values['partner_email'],
         }
+        # NOTE: this is rather shitty, because it means that one request minimum
+        # will be made at each rendering of the Stripe form, however this is how
+        # the checkout API work I'm afraid
```

Comment:
@dbo-odoo 
Checkout session will be created when you click on pay now button,  and after it will directly redirect to stripe.
so there no worries for multiple API calls.

Reply:
yes, you're right; i forgot the button is rendered only with the rpc call from the js payment_form

Reply:
comment to remove